From 9bfe22031475aed1712a74bbe452c8df317556b5 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:35 -0400
Subject: [tools] perf/config: Ignore generated files in feature-checks

Message-id: <1408953210-26343-208-git-send-email-jolsa@redhat.com>
Patchwork-id: 88850
O-Subject: [PATCH RHEL7.1 BZ1133083 207/322] perf config: Ignore generated files in feature-checks
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 56560ec692c142bb9ee404764e3b67999031ad19
Author: Chunwei Chen <tuxoko@gmail.com>
Date: Sat Dec 21 13:48:11 2013 +0800

description
===========
1. Rename the test-* binary files to test-*.bin for easier pattern matching as
   suggested by Ingo.
2. Ignore *.bin and *.d files.
---
 create mode 100644 tools/perf/config/feature-checks/.gitignore

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index 5a1f4df..14faeeb 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -126,7 +126,7 @@ endif
 
 feature_check = $(eval $(feature_check_code))
 define feature_check_code
-  feature-$(1) := $(shell $(MAKE) OUTPUT=$(OUTPUT_FEATURES) CFLAGS="$(EXTRA_CFLAGS) $(FEATURE_CHECK_CFLAGS-$(1))" LDFLAGS="$(LDFLAGS) $(FEATURE_CHECK_LDFLAGS-$(1))" -C config/feature-checks test-$1 >/dev/null 2>/dev/null && echo 1 || echo 0)
+  feature-$(1) := $(shell $(MAKE) OUTPUT=$(OUTPUT_FEATURES) CFLAGS="$(EXTRA_CFLAGS) $(FEATURE_CHECK_CFLAGS-$(1))" LDFLAGS="$(LDFLAGS) $(FEATURE_CHECK_LDFLAGS-$(1))" -C config/feature-checks test-$1.bin >/dev/null 2>/dev/null && echo 1 || echo 0)
 endef
 
 feature_set = $(eval $(feature_set_code))
@@ -173,7 +173,7 @@ CORE_FEATURE_TESTS =   \
 # to skip the print-out of the long features list if the file
 # existed before and after it was built:
 #
-ifeq ($(wildcard $(OUTPUT)config/feature-checks/test-all),)
+ifeq ($(wildcard $(OUTPUT)config/feature-checks/test-all.bin),)
   test-all-failed := 1
 else
   test-all-failed := 0
@@ -203,7 +203,7 @@ ifeq ($(feature-all), 1)
   #
   $(foreach feat,$(CORE_FEATURE_TESTS),$(call feature_set,$(feat)))
 else
-  $(shell $(MAKE) OUTPUT=$(OUTPUT_FEATURES) CFLAGS="$(EXTRA_CFLAGS)" LDFLAGS=$(LDFLAGS) -i -j -C config/feature-checks $(CORE_FEATURE_TESTS) >/dev/null 2>&1)
+  $(shell $(MAKE) OUTPUT=$(OUTPUT_FEATURES) CFLAGS="$(EXTRA_CFLAGS)" LDFLAGS=$(LDFLAGS) -i -j -C config/feature-checks $(addsuffix .bin,$(CORE_FEATURE_TESTS)) >/dev/null 2>&1)
   $(foreach feat,$(CORE_FEATURE_TESTS),$(call feature_check,$(feat)))
 endif
 
diff --git a/tools/perf/config/feature-checks/.gitignore b/tools/perf/config/feature-checks/.gitignore
new file mode 100644
index 0000000..80f3da0
--- /dev/null
+++ b/tools/perf/config/feature-checks/.gitignore
@@ -0,0 +1,2 @@
+*.d
+*.bin
diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index 19d1bc5..7cf6fcd 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -1,89 +1,90 @@
 
 FILES=     \
- test-all   \
- test-backtrace   \
- test-bionic   \
- test-dwarf   \
- test-fortify-source  \
- test-glibc   \
- test-gtk2   \
- test-gtk2-infobar  \
- test-hello   \
- test-libaudit   \
- test-libbfd   \
- test-liberty   \
- test-liberty-z   \
- test-cplus-demangle  \
- test-libelf   \
- test-libelf-getphdrnum  \
- test-libelf-mmap  \
- test-libnuma   \
- test-libperl   \
- test-libpython   \
- test-libpython-version  \
- test-libslang   \
- test-libunwind   \
- test-on-exit   \
- test-stackprotector-all  \
- test-timerfd
+ test-all.bin   \
+ test-backtrace.bin  \
+ test-bionic.bin   \
+ test-dwarf.bin   \
+ test-fortify-source.bin  \
+ test-glibc.bin   \
+ test-gtk2.bin   \
+ test-gtk2-infobar.bin  \
+ test-hello.bin   \
+ test-libaudit.bin  \
+ test-libbfd.bin   \
+ test-liberty.bin  \
+ test-liberty-z.bin  \
+ test-cplus-demangle.bin  \
+ test-libelf.bin   \
+ test-libelf-getphdrnum.bin \
+ test-libelf-mmap.bin  \
+ test-libnuma.bin  \
+ test-libperl.bin  \
+ test-libpython.bin  \
+ test-libpython-version.bin \
+ test-libslang.bin  \
+ test-libunwind.bin  \
+ test-libunwind-debug-frame.bin \
+ test-on-exit.bin  \
+ test-stackprotector-all.bin \
+ test-timerfd.bin
 
 CC := $(CC) -MD
 
 all: $(FILES)
 
-BUILD = $(CC) $(CFLAGS) -o $(OUTPUT)$@ $@.c $(LDFLAGS)
+BUILD = $(CC) $(CFLAGS) -o $(OUTPUT)$@ $(patsubst %.bin,%.c,$@) $(LDFLAGS)
 
 ###############################
 
-test-all:
+test-all.bin:
  $(BUILD) -Werror -fstack-protector-all -O2 -Werror -D_FORTIFY_SOURCE=2 -ldw -lelf -lnuma -lelf -laudit -I/usr/include/slang -lslang $(shell pkg-config --libs --cflags gtk+-2.0 2>/dev/null) $(FLAGS_PERL_EMBED) $(FLAGS_PYTHON_EMBED) -DPACKAGE='"perf"' -lbfd -ldl
 
-test-hello:
+test-hello.bin:
  $(BUILD)
 
-test-stackprotector-all:
+test-stackprotector-all.bin:
  $(BUILD) -Werror -fstack-protector-all
 
-test-fortify-source:
+test-fortify-source.bin:
  $(BUILD) -O2 -Werror -D_FORTIFY_SOURCE=2
 
-test-bionic:
+test-bionic.bin:
  $(BUILD)
 
-test-libelf:
+test-libelf.bin:
  $(BUILD) -lelf
 
-test-glibc:
+test-glibc.bin:
  $(BUILD)
 
-test-dwarf:
+test-dwarf.bin:
  $(BUILD) -ldw
 
-test-libelf-mmap:
+test-libelf-mmap.bin:
  $(BUILD) -lelf
 
-test-libelf-getphdrnum:
+test-libelf-getphdrnum.bin:
  $(BUILD) -lelf
 
-test-libnuma:
+test-libnuma.bin:
  $(BUILD) -lnuma
 
-test-libunwind:
+test-libunwind.bin:
  $(BUILD) -lelf
 
-test-libunwind-debug-frame:
+test-libunwind-debug-frame.bin:
  $(BUILD) -lelf
 
-test-libaudit:
+test-libaudit.bin:
  $(BUILD) -laudit
 
-test-libslang:
+test-libslang.bin:
  $(BUILD) -I/usr/include/slang -lslang
 
-test-gtk2:
+test-gtk2.bin:
  $(BUILD) $(shell pkg-config --libs --cflags gtk+-2.0 2>/dev/null)
 
-test-gtk2-infobar:
+test-gtk2-infobar.bin:
  $(BUILD) $(shell pkg-config --libs --cflags gtk+-2.0 2>/dev/null)
 
 grep-libs  = $(filter -l%,$(1))
@@ -95,7 +96,7 @@ PERL_EMBED_LIBADD = $(call grep-libs,$(PERL_EMBED_LDOPTS))
 PERL_EMBED_CCOPTS = `perl -MExtUtils::Embed -e ccopts 2>/dev/null`
 FLAGS_PERL_EMBED=$(PERL_EMBED_CCOPTS) $(PERL_EMBED_LDOPTS)
 
-test-libperl:
+test-libperl.bin:
  $(BUILD) $(FLAGS_PERL_EMBED)
 
 override PYTHON := python
@@ -112,31 +113,31 @@ PYTHON_EMBED_LIBADD = $(call grep-libs,$(PYTHON_EMBED_LDOPTS))
 PYTHON_EMBED_CCOPTS = $(shell $(PYTHON_CONFIG_SQ) --cflags 2>/dev/null)
 FLAGS_PYTHON_EMBED = $(PYTHON_EMBED_CCOPTS) $(PYTHON_EMBED_LDOPTS)
 
-test-libpython:
+test-libpython.bin:
  $(BUILD) $(FLAGS_PYTHON_EMBED)
 
-test-libpython-version:
+test-libpython-version.bin:
  $(BUILD) $(FLAGS_PYTHON_EMBED)
 
-test-libbfd:
+test-libbfd.bin:
  $(BUILD) -DPACKAGE='"perf"' -lbfd -ldl
 
-test-liberty:
+test-liberty.bin:
  $(CC) -o $(OUTPUT)$@ test-libbfd.c -DPACKAGE='"perf"' -lbfd -ldl -liberty
 
-test-liberty-z:
+test-liberty-z.bin:
  $(CC) -o $(OUTPUT)$@ test-libbfd.c -DPACKAGE='"perf"' -lbfd -ldl -liberty -lz
 
-test-cplus-demangle:
+test-cplus-demangle.bin:
  $(BUILD) -liberty
 
-test-on-exit:
+test-on-exit.bin:
  $(BUILD)
 
-test-backtrace:
+test-backtrace.bin:
  $(BUILD)
 
-test-timerfd:
+test-timerfd.bin:
  $(BUILD)
 
 -include *.d
-- 
1.7.1